### PR TITLE
Avoid PTv3 BEV projection NonZero export

### PR DESCRIPTION
**Pull request by mojomex**
## Summary

This PR adds an opt-in export path for the PTv3 detection BEV projector that skips BEV bounds-mask compaction when preprocessing already guarantees valid BEV grid coordinates.

The current guarded projection path exports boolean indexing as `NonZero` + `GatherND`, which TensorRT handles with data-dependent shape-host copies during detection-head inference. For the PTv3 multihead deployment configs, the point cloud is range-filtered/grid-sampled before the model, so the skip-stage BEV coordinates are expected to be inside the configured dense BEV canvas.

## Changes

- Add `assume_valid_grid_coord` to `PTv3BEVProjection`.
- Keep the default guarded behavior in the base BEV-neck config.
- Enable the unguarded export path for the PTv3 detection and multihead production configs.
- Add a focused test comparing guarded and unguarded projection outputs when all coordinates are valid.

## Benchmark Evidence

Measured on `sensing-bench` in isolation.

image

| Check | Baseline | Optimized artifact check |
|---|---|---|
| Variant | `ptv3-0710-pipe251exp-807-baseline` | `ptv3-0710-pipe251exp-807-det-bev-no-nonzero` |
| GPU total replicate mean | `38.860 ms` | `38.749 ms` |
| Inference replicate mean | `35.872 ms` | `35.590 ms` |
| Nsight det-head enqueue max (CPU-side!) | `12.946 ms` | `2.618 ms` |
| BEV shape-sync signature | `NonZero_3[size][DevicetoShapeHostCopy]` max `8.997 ms` | no `NonZero`, no `DevicetoShapeHostCopy`, and no `trainStation` |
| Det-head ONNX op count | current exported graph includes `NonZero`/`GatherND` | `NonZero=0`, `GatherND=0`, `ScatterElements=1` |

## Validation

- TensorRT det-head engine rebuild from the optimized ONNX
- PTv3 7x50 benchmark series
- Nsight Systems profile analysis

### CODE REVIEW

File: autoware_ml/tests/models/test_ptv3_detection.py [hunk 1]
```diff
@@ -19,6 +19,26 @@
...
+def test_ptv3_bev_projection_assume_valid_matches_guarded_path_for_valid_coords() -> None:
+    guarded = build_trans_model().bev_neck.bev_projector.eval()
+    unguarded = build_trans_model().bev_neck.bev_projector.eval()
+    unguarded.load_state_dict(guarded.state_dict())
+    unguarded.assume_valid_grid_coord = True
+
+    point_features = torch.randn(6, guarded.point_proj[0].in_features)
+    grid_coord = torch.tensor(
+        [[0, 0, 0], [1, 2, 0], [1, 2, 1], [3, 4, 0], [7, 7, 0], [5, 6, 2]],
+        dtype=torch.int32,
```
Comment (mojomex) [inline autoware_ml/tests/models/test_ptv3_detection.py:36]: Make BEV grid bounds explicit in this test, so the grid_coord can be easily human-verified to be within the bounds.

**Comment by amadeuszsz** [comment]
@codex review

File: autoware_ml/configs/tasks/detection3d/ptv3/voxel005_51m_nuscenes.yaml [hunk 2]
```diff
@@ -137,6 +137,7 @@ model:
   bev_neck:
     bev_projector:
       output_shape: [256, 256]
+      assume_valid_grid_coord: true
```
[the remark this answers is not in the document]
Comment (mojomex) [reply autoware_ml/configs/tasks/detection3d/ptv3/voxel005_51m_nuscenes.yaml:140]: Addressed in `046d264`: `PointsRangeFilter` now keeps points in `[lower, upper)` instead of `[lower, upper]`, matching the inference-side crop contract. Added transform coverage for exact max-bound points and a regression path through `GridSample` that asserts emitted `grid_coord` values remain inside the grid.

**Review by amadeuszsz — APPROVED** [review]
LGTM
